src/topology/algebra/order/lower_topology.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import topology.basic
+import topology.order
+import topology.separation
+import data.set.intervals.basic
+import order.upper_lower
+
+/-!
+# Lower topology
+
+This file introduces the lower topology on a preorder. It is shown that the lower topology on a
+partial order is T₀ and the non-empty complements of the upper closures of finite subsets form a
+basis.
+
+## Implementation notes
+
+Approach inspired by `order_topology` from topology.algebra.order.basic
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+
+variable (α : Type u)
+
+open set topological_space
+
+section pre_order
+
+variable [preorder α]
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[derive preorder, nolint unused_arguments]
+def lower := α
```

Comment:
This is way too vague as a name.

Reply:
@YaelDillies `lower_topology` instead?

Reply:
I'd suggest `with_lower_topology` (or maybe `with_lower` or `with_lower_top`)

Reply:
Should the namespace still be `lower_topology` - or do I change that to `with_lower_topology` too?